proxy_map.go

```diff
@@ -98,6 +98,83 @@ Once the lock context is created, it can be used to lock/unlock entries and used
 As mentioned before, lock context is a regular context.Context which carry a special lock ID.
 You can pass any context.Context to any Map function, but in that case lock ownership between operations using the same hazelcast.Client instance is not possible.
 
+Using the Near Cache
+
+Map entries in Hazelcast are partitioned across the cluster members.
+Hazelcast clients do not have local data at all.
+Suppose you read the key k a number of times from a Hazelcast client or k is owned by another member in your cluster.
+Then each map.Get(k) will be a remote operation, which creates a lot of network trips.
+If you have a data structure that is mostly read, then you should consider creating a local Near Cache, so that reads are sped up and less network traffic is created.
+
+These benefits do not come for free. See the following trade-offs:
+
+    * Members with a Near Cache has to hold the extra cached data, which increases memory consumption.
+    * If invalidation is enabled and entries are updated frequently, then invalidations will be costly.
+    * Near Cache breaks the strong consistency guarantees; you might be reading stale data.
+
+Near Cache is highly recommended for data structures that are mostly read.
+
+You must enable the Near Cache on the client, without the need to configure it on the server.
+Note that Near Cache configuration is specific to the server or client itself.
+A data structure on a server may not have Near Cache configured while the same data structure on a client may have Near Cache configured.
+They also can have different Near Cache configurations.
+
+If you are using the Near Cache, you should take into account that your hits to the keys in the Near Cache are not reflected as hits to the original keys on the primary members.
+This has for example an impact on Map's maximum idle seconds or time-to-live seconds expiration.
+Therefore, even though there is a hit on a key in Near Cache, your original key on the primary member may expire.
+
+Note: Near Cache works only when you access data via map.Get(k).
+Data returned using a predicate is not stored in the Near Cache.
```

Comment:
```suggestion
Data returned using a predicate or SQL is not stored in the Near Cache.
```
I think we should add SQL too

Reply:
https://github.com/hazelcast/hazelcast-go-client/pull/845/commits/262c5cf8f465d87b6c0fa06c1fb59ed656c7745e